### PR TITLE
Add autofill for flashcard label

### LINKED ISSUE #205: Flashcards: Labels shouldn't be required (or automatically filled in) for 2-sided flashcards

### PR DESCRIPTION
**Pull request by mardom1**
# Description of changes

Closes #205 

## How has this been tested?

Please describe the test strategy you followed.

- [ ] automated unit test
- [ ] automated integration test
- [ ] automated acceptance test
- [ ] manual, exploratory test

In case of manual test, please document the test well including a set of user instructions and prerequisites. Each including an action, it's result, and where appropriate a screenshot.

## Checklist before requesting a review

- [ ] My code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My code fulfilles all acceptance criteria
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have made corresponding changes to the documentation
- [ ] I have added explanation of architectural decision and rationales to [wiki/adr](https://github.com/IT-REX-Platform/wiki/tree/main/adr)
- [ ] I have updated the changes in the ticket description

## Checklist for reviewer

- [ ] The code works and does not throw errors
- [ ] The code is easy to understand and there are no confusing parts
- [ ] The code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [ ] The code change accomplishes what it is supposed to do
- [ ] I cannot think of any use case in which the code does not behave as intended
- [ ] The added and existing tests reasonably cover the code change
- [ ] I cannot think of any test cases, input or edge cases that should be tested in addition
- [ ] Description of the change is included in the documentation

### CODE REVIEW

File: app/courses/[courseId]/flashcards/[flashcardSetId]/lecturer.tsx [hunk 1]
```diff
@@ -540,16 +541,59 @@ function EditSideModal({
...
+      return;
+    }
+
+    // Auto-populate label
+    if (isQuestion && !isAnswer && label !== "Question") {
+      setLabel("Question");
+    } else if (!isQuestion && isAnswer && label !== "Answer") {
+      setLabel("Answer");
+    } else if (isQuestion === isAnswer) {
+      setLabel("");
```
Comment (SkyDiverStar) [inline app/courses/[courseId]/flashcards/[flashcardSetId]/lecturer.tsx:567]: Warum leer? Wenn beides "true" ist, könnte man doch "Question and Answer" oder "QnA" oder Q&A" setzen. Da Flashcards beides gleichzeitig sein können.

**Comment by SkyDiverStar** [comment]
Ansonsten sieht alles ok aus, sehe sonst keine Probleme

File: app/courses/[courseId]/flashcards/[flashcardSetId]/lecturer.tsx (hunk 1 above)
Comment (mardom1) [reply app/courses/[courseId]/flashcards/[flashcardSetId]/lecturer.tsx:567]: Wenn beide true sind, kann das aus mehreren Gründen sein und braucht wahrscheinlich einen deskriptiven Titel, deshalb wird das Feld gecleared. Hier geht es ja hauptsächlich darum den Standardfall zu vereinbaren mit einer Frage und einer Antwort.
